clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -143,9 +143,12 @@ private boolean matchesRequestedPartitions(FetchRequest request, FetchResponse r
     /**
      * Set-up a fetch request for any node that we have assigned partitions for which doesn't already have
      * an in-flight fetch or pending fetch data.
+     * @return number of fetches sent
      */
-    public void sendFetches() {
+    public int sendFetches() {
+        int fetches = 0;
```

Comment:
It doesn't seem like we need this variable. Why not assign the result of `createFetchRequests` and call `size` on it? Let mutable variables that need to be tracked.